### ISSUE TITLE
XLogOpenRelation is marked to be removed. Can we?

### ISSUE THREAD

**Issue by Eulerizeit** [greenplum-db/gpdb#3755]
Possibly related to #3754.
 ```
/*
    GPDB_84_MERGE_FIXME: this block of code (brought over from
    XLogOpenRelation) was marked to be removed.  Can we?
    Is it related to filerep?
*/
 // UNDONE: Can't remove this block of code yet until boot time calls to this routine are analyzed…

 GPDB_84_MERGE_FIXME: this if block was moved from the
 removed XLogOpenRelation(). Is this the correct place for it?
 What does it do?
```

[./src/backend/access/transam/xlogutils.c:322](https://github.com/greenplum-db/gpdb-postgres-merge/blob/iteration_3_redux/src/backend/access/transam/xlogutils.c#L322)
[./src/backend/access/transam/xlogutils.c:446](https://github.com/greenplum-db/gpdb-postgres-merge/blob/iteration_3_redux/src/backend/access/transam/xlogutils.c#L446)

**Comment by jchampio** [comment]
Regarding the first FIXME: we can't remove that code until filerep is gone, *and* we need to update it to handle AO tables as well.

For the second FIXME, the directory creation code in question seems to be duplicated in multiple places. Since tests are passing now, we're not sure if it's necessary. It seems to only apply to a standby master, when using non-default filespaces. We haven't found a good test case to reproduce the initial issue that it was supposed to solve -- it seems like originally, there was a problem where the filespace hashtable was not initialized on the standby.

**Comment by asimrp** [comment]
This issue is addressed by PR #3842.  See issue #3843 for the handling necessary for appendonly tables.
